app/controllers/submissions_controller.rb

```diff
@@ -167,14 +167,13 @@ def manually_collect_and_begin_grading
     end
 
     @revision_identifier = params[:current_revision_identifier]
-    apply_late_penalty = if params[:apply_late_penalty].nil?
-                           false
-                         else
-                           params[:apply_late_penalty]
-                         end
+    apply_late_penalty = params[:apply_late_penalty] == 'true'
+    retain_existing_grading = params[:retain_existing_grading] == 'true'
```

Comment:
I'm not sure if you manually tested this after making this change, but it seems like the parameter value is `"on"`, not `"true"`, which is a bit strange.

Reply:
Ah whoops, I did not manually test this in this iteration of changes - sorry about that! However, I recall now that I actually tried to make this change a couple of weeks ago when messing around with this, and I did run into this problem of it being `on` instead of `true`, which is why I left it the way it was. I will go ahead and revert this change for now - thanks for the catch.